x-pack/plugin/stack/src/main/java/org/elasticsearch/xpack/stack/StackTemplateRegistry.java

```diff
@@ -132,53 +131,10 @@ public StackTemplateRegistry(
         this.clusterService = clusterService;
         this.featureService = featureService;
         this.stackTemplateEnabled = STACK_TEMPLATES_ENABLED.get(nodeSettings);
-        this.logsIndexModeTemplateEnabled = CLUSTER_LOGSDB_ENABLED.get(nodeSettings);
-    }
-
-    @Override
-    public void initialize() {
-        super.initialize();
-        clusterService.getClusterSettings().addSettingsUpdateConsumer(STACK_TEMPLATES_ENABLED, this::updateEnabledSetting);
-    }
-
-    private void updateEnabledSetting(boolean newValue) {
-        if (newValue) {
-            this.stackTemplateEnabled = true;
-        } else {
-            logger.info(
-                "stack composable templates [{}] and component templates [{}] will not be installed or reinstalled",
-                String.join(",", getComposableTemplateConfigs().keySet()),
-                String.join(",", getComponentTemplateConfigs().keySet())
-            );
-            this.stackTemplateEnabled = false;
-        }
-    }
-
-    private static final List<LifecyclePolicyConfig> LIFECYCLE_POLICY_CONFIGS = List.of(
-        new LifecyclePolicyConfig(LOGS_ILM_POLICY_NAME, "/logs@lifecycle.json", ADDITIONAL_TEMPLATE_VARIABLES),
-        new LifecyclePolicyConfig(METRICS_ILM_POLICY_NAME, "/metrics@lifecycle.json", ADDITIONAL_TEMPLATE_VARIABLES),
-        new LifecyclePolicyConfig(SYNTHETICS_ILM_POLICY_NAME, "/synthetics@lifecycle.json", ADDITIONAL_TEMPLATE_VARIABLES),
-        new LifecyclePolicyConfig(ILM_7_DAYS_POLICY_NAME, "/7-days@lifecycle.json", ADDITIONAL_TEMPLATE_VARIABLES),
-        new LifecyclePolicyConfig(ILM_30_DAYS_POLICY_NAME, "/30-days@lifecycle.json", ADDITIONAL_TEMPLATE_VARIABLES),
-        new LifecyclePolicyConfig(ILM_90_DAYS_POLICY_NAME, "/90-days@lifecycle.json", ADDITIONAL_TEMPLATE_VARIABLES),
-        new LifecyclePolicyConfig(ILM_180_DAYS_POLICY_NAME, "/180-days@lifecycle.json", ADDITIONAL_TEMPLATE_VARIABLES),
-        new LifecyclePolicyConfig(ILM_365_DAYS_POLICY_NAME, "/365-days@lifecycle.json", ADDITIONAL_TEMPLATE_VARIABLES)
-    );
-
-    @Override
-    protected List<LifecyclePolicyConfig> getLifecycleConfigs() {
-        return LIFECYCLE_POLICY_CONFIGS;
-    }
-
-    @Override
-    protected List<LifecyclePolicy> getLifecyclePolicies() {
-        return lifecyclePolicies;
+        this.componentTemplateConfigs = loadComponentTemplateConfigs(CLUSTER_LOGSDB_ENABLED.get(nodeSettings));
```

Comment:
I didn't share the method to load configs between the two registries because they load slightly different things.

Reply:
Also I kept the `CLUSTER_LOGSDB_ENABLED` setting in `StackTemplateRegistry` even if it is used by `LegacyStackTemplateRegistry` because the latter one is deprecated.

Reply:
This is not used anymore by `LegacyStackTemplateRegistry` after injecting always "standard" for legacy templates.

Reply:
LegacyStackTemplateRegistry looks good now.